### PR TITLE
[BETA]feat: support quickapp compiler

### PR DESCRIPTION
**Pull request by lovesong**
*Before* submitting a pull request, please make sure the following is done...

1. Fork the repo and create your branch from `master`.
2. If you've added code that should be tested, add tests!
3. If you've changed APIs, update the documentation.
4. Ensure the test suite passes (`npm test`).
5. Make sure your code lints (`npm run lint`) - we've done our best to make sure these rules match our internal linting guidelines.

### CODE REVIEW

File: packages/jsx2qa-compiler/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+output:
+
+- ast
+  - Babel 7 format AST of JS code
+- imported
+  - Imported modules and local identifiers
+- exported
+  - Exported identifiers
+- template
+  - axml template for miniapp
```
Comment (SoloJiang) [inline packages/jsx2qa-compiler/README.md:43]: 不太对

File: packages/jsx2qa-compiler/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  },
+  exported: ["default"],
+  code:
+    'import { createComponent as __create_component__, Component as __component__ } from "jsx2mp-runtime";\n\nconst __def__ = class extends __component__ {\n  render() {\n    return {};\n  }\n\n};\n\nComponent(__create_component__(__def__, {\n  events: []\n}));',
+  map: null,
+  config: {
+    component: true
+  },
+  style: "",
+  usingComponents: {},
```
Comment (SoloJiang) [inline packages/jsx2qa-compiler/README.md:78]: +1

File: packages/jsx2qa-compiler/README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  importComponents: {},
+  iconfontMap: [],
+  template: "<view>hello world</view>"
```
Comment (SoloJiang) [inline packages/jsx2qa-compiler/README.md:81]: +1

File: packages/jsx2qa-compiler/package.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "jsx2qa-compiler",
+  "version": "1.0.0",
```
Comment (SoloJiang) [inline packages/jsx2qa-compiler/package.json:3]: 0.1.0

File: packages/jsx2qa-compiler/src/adapter.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  },
+  wechat: {
+    onClick: 'bindtap',
+    onLongPress: 'bindlongpress',
+    onTouchStart: 'bindtouchstart',
+    onTouchEnd: 'bindtouchend',
+    onTouchMove: 'bindtouchmove',
+    onTouchCancel: 'bindtouchcancel',
+  }
+};
```
Comment (SoloJiang) [inline packages/jsx2qa-compiler/src/adapter.js:13]: remove

File: packages/jsx2qa-compiler/src/modules/__tests__/jsx-plus.js [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+      const index = 'index' + count++;
+      expect(genExpression(ast))
+        .toEqual(`<View>
+        <View for={array.map((val, ${index}) => {
+    return {
+      val: val,
+      ${index}: ${index},
+      d0: format(val)
+    };
+  })} a:for-item="val" a:for-index="${index}">{{
```
Comment (SoloJiang) [inline packages/jsx2qa-compiler/src/modules/__tests__/jsx-plus.js:137]: a:for-item ?

File: packages/jsx2qa-compiler/src/modules/components.js [hunk 7]
```diff
@@ -0,0 +1,530 @@
...
+      parsed.componentDependentProps = {};
+    }
+    if (!parsed.iconfontMap) {
+      parsed.iconfontMap = [];
+    }
+    const { contextList, dynamicValue, componentsAlias, iconfontMap } = transformComponents(parsed, options);
+    transformDataset(parsed, options);
+    // Collect used components
+    Object.keys(componentsAlias).forEach(componentTag => {
+      if (!parsed.usingComponents) {
```
Comment (SoloJiang) [inline packages/jsx2qa-compiler/src/modules/components.js:358]: usingComponents 的代码还需要吗

File: packages/jsx2qa-compiler/src/modules/element.js [hunk 8]
```diff
@@ -0,0 +1,911 @@
...
+
+function isNativeComponent(path, platform) {
+  const {
+    node: { name: tagName }
+  } = path.parentPath.get('name');
+  console.log('tagName', tagName);
+  return !!getCompiledComponents(platform)[tagName];
+}
+
+// function transformPreComponentAttr(ast) {
```
Comment (SoloJiang) [inline packages/jsx2qa-compiler/src/modules/element.js:573]: 注释去掉

File: packages/jsx2qa-compiler/src/modules/components.js (hunk 7 above)
Comment (lovesong) [reply packages/jsx2qa-compiler/src/modules/components.js]: 可以去掉

File: packages/jsx2qa-compiler/src/getCompiledComponents.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+module.exports = function(platform) {
+  switch (platform) {
+    case 'quickapp':
+      return {
+        'rax-view': 'div',
+      };
+  }
+};
```
Comment (SoloJiang) [inline packages/jsx2qa-compiler/src/getCompiledComponents.js:8]: 这个地方没必要用函数了吧

File: packages/jsx2qa-compiler/src/index.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+const { parse } = require('./parser');
+const { baseOptions } = require('./options');
+const adapter = require('./adapter');
+
+/**
+ * @param template {String} Template string.
+ * @param options {Object} Compiler options.
+ */
+function compile(template, options) {
+  const type = options.platform && options.platform.type || 'ali';
```
Comment (SoloJiang) [inline packages/jsx2qa-compiler/src/index.js:11]: 还有这里，我觉得整个 adapter 的机制都不需要了

**Pull request closed without merging**